### PR TITLE
Create endpoint to update data under curation

### PR DESCRIPTION
**Pull request by dglemos**
G2P-122

### CODE REVIEW

File: gene2phenotype_project/gene2phenotype_app/views.py [hunk 1]
```diff
@@ -850,3 +851,36 @@ def list(self, request, *args, **kwargs):
...
+        curation_obj = self.get_queryset().first()
+
+        serializer = CurationDataSerializer(curation_obj, data=request.data, context={'user': self.request.user})
+
+        if serializer.is_valid():
+            serializer.save()
+            return Response({"message": "Data updated successfully"})
+
+        else:
+            return Response({"message": "failed", "details": serializer.errors})
```
[the remark this answers is not in the document]
Comment (dglemos) [reply gene2phenotype_project/gene2phenotype_app/views.py]: Updated

File: gene2phenotype_project/gene2phenotype_app/views.py [hunk 2]
```diff
@@ -850,3 +851,51 @@ def list(self, request, *args, **kwargs):
...
+        try:
+            print(request.data)
+            validate(instance=request.data, schema=schema)
+        except exceptions.ValidationError as e:
+            return Response({"message": "JSON data does not follow the required format, Required format is" + str(e)}, status=status.HTTP_400_BAD_REQUEST)
+
+        serializer = CurationDataSerializer(curation_obj, data=request.data, context={'user': self.request.user})
+
+        if serializer.is_valid():
+            serializer.save()
```
Comment (olaaustine) [inline gene2phenotype_project/gene2phenotype_app/views.py:897]: Suggestion - I think this should be `serializer.update(curation_obj, request.data)`, serializer.save should be for create, I mean both of them work but this is the update bit so should use the update function of the serializer

File: gene2phenotype_project/gene2phenotype_project/settings.py [hunk 3]
```diff
@@ -125,7 +125,7 @@
 
 LANGUAGE_CODE = 'en-us'
 
-TIME_ZONE = 'UTC'
+TIME_ZONE = 'Europe/London'
```
[the remark this answers is not in the document]
Comment (olaaustine) [reply gene2phenotype_project/gene2phenotype_project/settings.py:128]: Will be done in the next PR

**Review by olaaustine — COMMENTED** [review]
Just some minor changes. Thanks @dglemos

File: gene2phenotype_project/gene2phenotype_app/views.py (hunk 2 above)
Comment (olaaustine) [reply gene2phenotype_project/gene2phenotype_app/views.py:897]: Actually considering [this ](https://www.django-rest-framework.org/api-guide/serializers/#saving-instances:~:text=Calling%20.save()%20will%20either%20create%20a%20new%20instance%2C%20or%20update%20an%20existing%20instance%2C%20depending%20on%20if%20an%20existing%20instance%20was%20passed%20when%20instantiating%20the%20serializer%20class%3A), please ignore

File: gene2phenotype_project/gene2phenotype_app/serializers.py [hunk 4]
```diff
@@ -1155,7 +1154,14 @@ def create(self, validated_data):
 
         return new_curation_data
 
+    @transaction.atomic
+    def update(self, instance, validated_data):
+        instance.json_data = validated_data['json_data']
+        instance.date_last_update = timezone.now()
+        instance.save()
+
+        return super().update(instance, validated_data)
```
[the remark this answers is not in the document]
Comment (dglemos) [reply gene2phenotype_project/gene2phenotype_app/serializers.py]: I just need to return the instance.
Updated.

**Pull request merged**